react/features/video-menu/components/web/FlipLocalVideoButton.js

```diff
@@ -0,0 +1,103 @@
+/* @flow */
+
+import React, { Component } from 'react';
+
+import { translate } from '../../../base/i18n';
+import { connect } from '../../../base/redux';
+import { updateSettings } from '../../../base/settings';
+
+import VideoMenuButton from './VideoMenuButton';
+
+/**
+ * The type of the React {@code Component} props of {@link FlipLocalVideoButton}.
+ */
+type Props = {
+
+    /**
+     * The current local flip x status.
+     */
+    _localFlipX: boolean,
+
+    /**
+     * The redux dispatch function.
+     */
+    dispatch: Function,
+
+    /**
+     * Invoked to obtain translated strings.
+     */
+    t: Function
+};
+
+/**
+ * Implements a React {@link Component} which displays a button for flipping the local viedo.
+ *
+ * @extends Component
+ */
+class FlipLocalVideoButton extends Component<Props> {
```

Comment:
This looks like a good candidate for a functional component. If not, it should inherit from `PureComponent` at least.